packages/browser-destinations/src/destinations/heap/trackEvent/index.ts

```diff
@@ -75,38 +75,54 @@ const action: BrowserActionDefinition<Settings, HeapApi, Payload> = {
     const eventName = event.payload.name
     const browserArrayLimit = event.settings.browserArrayLimit || 0
     const browserArrayLimitSet = !!browserArrayLimit
-    let arrayEventsCount = 0
 
-    for (const [key, value] of Object.entries(eventProperties)) {
-      if (browserArrayLimitSet && arrayEventsCount >= browserArrayLimit) {
-        break
-      }
+    if (browserArrayLimitSet) {
+      eventProperties = heapTrackArrays(heap, eventName, eventProperties, browserArrayLimit)
```

Comment:
Diff is ugly but most of it is moving the functionality to `heapTrackArrays` (naming suggestions welcomed)